js/bundle.go

```diff
@@ -169,6 +155,7 @@ func (b *Bundle) MakeArchive() *lib.Archive {
 		Filename: b.Filename,
 		Data:     []byte(b.Source),
 		Pwd:      b.BaseInitContext.pwd,
+		Env:      b.Env,
```

Comment:
Sounds goods, that alleviates my concerns about potential privacy issues with running `k6 archive` and `k6 cloud`. The only issue remaining then would be the different default behavior between `run` and the other commands. I'm assuming that `inspect` should behave like `archive` and `cloud` and use only explicitly passed variables with `--env`/`-e` while ignoring the real system env by default. 

To minimize the impact from the different defaults, I'm considering flipping the `--no-system-env-vars` to be called `--use-system-env-vars` with default true for `run` and false for `cloud`/`archive`/`inspect`. What do you think? It seems to me that would be easier to reason about than the double negatives. Right now with `--no-system-env-vars` it's "by default don't *not include* system env vars when running tests locally and do *not include* them when running in the cloud". 

I'm planning to implement the fixes this evening or tomorrow, I'll ping you when they're ready for a new review.

Reply:
Agree, flipping `--no-system-env-vars` to `--use-system-env-vars` (with a default `true` for `run` command) makes sense. Given that the flag will now need to take a boolean (to support turning off for `run`/`inspect` and turning on for `cloud`/`archive`) maybe rename it to just `--system-env-vars` (or `--include-system-env-vars`)? Are we going to accept "true"|"false" and/or 0|1 as values?

I do think we should include system env vars by default for `inspect` though since it's run locally like `run`.

Reply:
Not sure how the [pflag](https://github.com/spf13/pflag) library deals with boolean/set values, I'll check and test it later. Since we'll have different default values for different commands, I'll make sure that it's possible to explicitly specify `true`/`false` ( and/or `0`/`1`, `enabled/disabled`, etc.). I assume that it should be able to accept values, if not, I can just use a string flag and parse it. I'll also add both cases to the end-to-end tests to make sure everything's OK.

Regarding the name, I think `--system-env-vars` would be best if the expected values are `enabled`/`disabled`, but `--use-system-env-vars`/`--include-system-env-vars` sound better for `true`/`false` values.

For the `inspect` command - yeah, it's run locally, but I assume that the most common use case is inspecting tests and `archive`s which may be executed on your `cloud` or on a private CI/CD system somewhere. If that's the case, I don't think that evaluating the user's system environment variables by default is the best strategy when `inspect`ing. That said, I'm far from knowledgeable about all of the k6 use cases, so if you think `inspect` should behave like `run` with regards to environment variables, that's what I'll do.

Reply:
Ok, agree on the naming of the flag. Use the one that fits with the expected value.

You're right, didn't think of the archive case for `inspect`. Ok, default to excluding system env vars here.